app/scripts/lib/fxa-client.js

```diff
@@ -278,6 +278,35 @@ function (_, FxaClient, $, xhr, p, Session, AuthErrors, Constants) {
         });
     },
 
+    completeAccountUnlock: function (uid, code) {
+      return this._getClient()
+        .then(function (client) {
+          return client.accountUnlockVerifyCode(uid, code);
+        });
+    },
+
+    sendAccountUnlockEmail: function (email, relier) {
+      var self = this;
+      return self._getClient()
+        .then(function (client) {
+          if (self._accountUnlockResendCount >= Constants.ACCOUNT_UNLOCK_RESEND_MAX_TRIES) {
+            return p(true);
```

Comment:
>  Could we consider surfacing some error to the user in this case?

The silent success was intentional to give fewer clues to casual spammers to just reload the page and starting over. 

Unfortunately, email rate limiting code exists in two places and needs to be unified. Ref #1988